### PR TITLE
BNR - Enable BNR on device change

### PR DESCRIPTION
**Pull request by mkesavan13**
# COMPLETES [SPARK-346779](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-346779)

## This pull request addresses

The BNR is not enabled when a device changes. This pull request addresses that.

## by making the following changes

- Enabling BNR on updateAudio by introducing various BNR states. 
- Correcting tests appropriately

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by arun3528 — COMMENTED** [review]
Business logic looks ok

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 1]
```diff
@@ -5744,6 +5758,38 @@ export default class Meeting extends StatelessWebexPlugin {
...
+    }
+    catch (error) {
+      LoggerProxy.logger.error('Meeting:index#internal_enableBNR.', error);
+      throw error;
+    }
+  }
+
   /**
    * enableBNR API
    * @returns {Promise<Boolean>}
```
Comment (meta-dipanshu-sharma) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:5795]: I guess we're returning a Boolean value `isSuccess` not a Promise<Boolean>. Correct me if I'm missing something.

**Review by meta-dipanshu-sharma — CHANGES_REQUESTED** [review]

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 2]
```diff
@@ -5758,32 +5804,37 @@ export default class Meeting extends StatelessWebexPlugin {
...
+      else if (this.isAudioMuted()) {
+        throw new Error('Cannot enable BNR while meeting is muted');
+      }
+
+
+      this.bnrStatus = BNR_STATUS.SHOULD_ENABLE;
+
       const audioStream = MediaUtil.createMediaStream([this.mediaProperties.audioTrack]);
 
       LoggerProxy.logger.info('Meeting:index#enableBNR. BNR enabled track obtained from WebRTC & sent to updateAudio');
```
[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Sorry missed this one out. Updated this.

[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Thanks for pointing out @meta-dipanshu-sharma

**Review by arun3528 — APPROVED** [review]
Looks good to me , i thought i already review the code 

fix the package.lock file

**Comment by mkesavan13** [comment]
> Looks good to me , i thought i already review the code
> 
> fix the package.lock file

Have replaced the entire lock file. Not sure if it is the right way. Could you please check and let me know if that is okay @arun3528 ?

**Comment by arun3528** [comment]
Tested locally and samples passed , Unit tests are fixed by rajesh

**Pull request merged**